File: src/main/java/com/melonbase/microquark/microstream/MicroStreamLifecycleBean.java
package com.melonbase.microquark.microstream;

import io.quarkus.runtime.ShutdownEvent;
import io.quarkus.runtime.StartupEvent;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import javax.enterprise.context.ApplicationScoped;
import javax.enterprise.event.Observes;
import javax.inject.Inject;

@ApplicationScoped
public class MicroStreamLifecycleBean {

  private static final Logger LOG = LoggerFactory.getLogger(MicroStreamLifecycleBean.class);

  @Inject
  MicroStreamStorage storage;

  void onStart(@Observes StartupEvent ev) {
    LOG.info("microstream storage active: {}", storage.getStorage().isActive());
    LOG.info("microstream storage running: {}", storage.getStorage().isRunning());
  }

  void onStop(@Observes ShutdownEvent ev) {
    LOG.info("Shutting down microstream storage.");

    storage.shutdown();

    LOG.info("microstream storage successfully shut down.");
  }
}

File: src/main/java/com/melonbase/microquark/microstream/MicroStreamLifecycleBean.kt
package com.melonbase.microquark.microstream

import io.quarkus.runtime.ShutdownEvent
import io.quarkus.runtime.StartupEvent
import mu.KotlinLogging
import one.microstream.storage.types.StorageManager
import javax.enterprise.context.ApplicationScoped
import javax.enterprise.event.Observes
import javax.inject.Inject

private val log = KotlinLogging.logger {}

@ApplicationScoped
class MicroStreamLifecycleBean @Inject constructor(val storage: StorageManager) {

  fun onStart(@Observes ev: StartupEvent) {
    log.info("MicroStream storage running: {}", storage.isRunning)
  }

  fun onStop(@Observes ev: ShutdownEvent) {
    log.info("Shutting down MicroStream storage.")
    storage.shutdown()
    log.info("MicroStream storage successfully shut down.")
  }
}